housekeeper/store/filters/file_join_filters.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+from typing import Callable, List, Optional
+from sqlalchemy.orm import Query
+from sqlalchemy import func as sqlalchemy_func
+
+from housekeeper.store.models import File, Tag
+
+
+def filter_files_by_tags(files_query: Query, tag_names: List[str], **kwargs) -> Query:
+    """Filter files by tag names."""
+    return (
+        files_query.filter(Tag.name.in_(tag_names))
+        .group_by(File.id)
+        .having(sqlalchemy_func.count(Tag.name) == len(tag_names))
+    )
+
+
+def filter_files_by_archive(files_query: Query, is_archived: bool, **kwargs) -> Query:
+    """Filters the query depending on if the files are archived or not."""
+    return files_query.filter((File.archive != None) == is_archived)
+
+
+class FileJoinFilter(Enum):
+    """Define filter functions for joins between File and other tables."""
+
+    FILTER_FILES_BY_TAGS: Callable = filter_files_by_tags
+    FILTER_FILES_BY_ARCHIVE: Callable = filter_files_by_archive
+
+
+def apply_file_join_filter(
```

Comment:
```suggestion
def apply_file_filter(
```

Reply:
Well there is already an `apply_file_filter` method for only file queries. This applies filters for file queries with other tables (tag, archive) joined. Any other name suggestion?